test/src/main/java/org/apache/accumulo/test/ShellServerIT.java

```diff
@@ -1431,18 +1436,20 @@ public void help() throws Exception {
     }
   }
 
-  // TODO - evaluate this test is testing what is expected or history is working.
   @Test
   public void history() throws Exception {
-    final String table = name.getMethodName();
+    final String table = getUniqueNames(1)[0];
 
-    ts.exec("history -c", true);
-    ts.exec("createtable " + table);
-    ts.exec("deletetable -f " + table);
-    // TODO - this may be passing only because method name is history.
+    // test clear history command works
+    ts.writeToHistory("foo");
+    ts.exec("history -c", true, "foo", false);
+
+    // write to history file. Does not execute functions
+    ts.writeToHistory("createtable " + table);
+    ts.writeToHistory("deletetable -f " + table);
+
+    // test that history command prints contents of history file
     ts.exec("history", true, table, true);
-    // TODO - what is this testing?
-    ts.exec("history", true, "history", true);
```

Comment:
I think this was just a basic test ensuring that exec caused the command to be added to the history. If that doesn't work in the test case, because of the way the test is written, that's fine to remove this, so long as executing commands in the shell still appends to the history.

Reply:
In a previous iteration, I kept this line but replaced the expected string with "foo". That is now being used in the validity check so I removed it here. The rest of what you said is still true and with your additions above, I believe the history command is thoroughly tested.